cmd/elementGen/elements.go

```diff
@@ -0,0 +1,727 @@
+/*
+ * MIT LICENSE
+ *
+ * Copyright © 2018, G.Ralph Kuntz, MD.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"go/format"
+	"os"
+	"path/filepath"
+	"sort"
+	"strings"
+	"text/template"
+)
+
+type (
+	// Desc contains a description of each HTML element that should be generated.
+	// - Name is the Pascal-case version (e.g. BaseFont).
+	// - JS is the lowercase version (e.g. basefont).
+	// - Override is used if the JS version is not the lowercase of the name (e.g. Script -> s).
+	// - Basic indicates that in "honnef.co/go/js/dom", we should use dom.BasicHTMLElement instead of
+	//   HTMLXXXElement, where XXX is the element name (e.g. dom.HTMLAppletElement).
+	// - DOM is the selected "honnef.co/go/js/dom" name (e.g. either dom.BasicHTMLElement or
+	//   dom.HTMLAppletElement).
+	Desc struct {
+		Name       string
+		JS         string
+		Override   string
+		Attributes []Attr
+		Basic      bool // True if honnef.co/go/js/dom declares a specific Element type.
```

Comment:
Can I suggest we invert the sense of this field? Because `Basic: true` appears in most elements below. Instead, invert it to by `NonBasic` (or some such) and set that to `true` when `Basic` would otherwise be `false`. Means the definitions shorten significantly below.